### PR TITLE
Http

### PR DESCRIPTION
**Pull request by 1verapoles**
Task https://github.com/rolling-scopes-school/tasks/blob/master/tasks/angular/rxjs-observables-http.md
21.08.2021/22.08.2021
Score 100

### CODE REVIEW

File: src/app/core/components/header/header.component.ts [hunk 1]
```diff
@@ -9,11 +11,24 @@ import { YoutubeService } from 'src/app/youtube/services/youtube.service';
...
 
-  onSubmit(event: any) {
-    event.preventDefault();
-    this.youtubeService.formSubmitted();
+  onKeyUp(e: any) {
+    fromEvent(e.target, 'keyup')
+      .pipe(
+        map((input: any) => input.target.value),
+        map((value: string) => value.trim()),
+        filter((query: string) => query.length >= 3),
```
Comment (pavel-mentor-account) [inline src/app/core/components/header/header.component.ts]: what is 3?. Please extract it to constant

File: src/app/core/components/header/header.component.ts [hunk 2]
```diff
@@ -9,11 +11,24 @@ import { YoutubeService } from 'src/app/youtube/services/youtube.service';
 })
 export class HeaderComponent {
 
-  constructor(private youtubeService: YoutubeService, private loginService: LoginService) { }
+  constructor(private youtubeService: YoutubeService, public loginService: LoginService) { }
 
-  onSubmit(event: any) {
-    event.preventDefault();
-    this.youtubeService.formSubmitted();
+  onKeyUp(e: any) {
```
Comment (pavel-mentor-account) [inline src/app/core/components/header/header.component.ts]: please add in all methods type which method returns

**Review by pavel-mentor-account — COMMENTED** [review]
everything works, except that instead of two API calls application make 6 calls. which is wrong

File: src/app/core/components/header/header.component.ts (hunk 1 above)
Comment (1verapoles) [reply src/app/core/components/header/header.component.ts]: I extracted it to constant

File: src/app/core/components/header/header.component.ts (hunk 2 above)
Comment (1verapoles) [reply src/app/core/components/header/header.component.ts]: I added types

**Pull request merged**